### PR TITLE
Add jobs and scripts for precompilation

### PR DESCRIPTION
**Pull request by Noarkhh**
Add jobs for building and releasing precompiled packages

### CODE REVIEW

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:fd1e91d9c47db49e1f697b49990b32938d8b8e22

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:dd0035ccffa80b9956b8bb8e5dabb4e8cd57fdf9

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:c014df375f40b40bbe64eb4375d67b08a5fd1e66

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:f8799b932221fcc9db4a83639fc67d42e850fd21

File: src/jobs/publish_precompiled.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+description: >
+  Publish a precompiled package
+
+parameters:
+  version:
+    description: "Version of the github release"
+    type: "string"
```
Comment (varsill) [inline src/jobs/publish_precompiled.yml:7]: [NIT] To keep it consistent with other parameters descriptions, we can remove the quotes there and move the `docker:` part right below the `description`

File: src/scripts/precompile_for_linux.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+artifact_name="${PACKAGE_NAME}_linux"
+apt-get update
+apt-get install -y curl gcc git make g++ bzip2
+mkdir -p ~/project/workspace/$artifact_name/include
+mkdir -p ~/project/workspace/$artifact_name/lib
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+export PATH="/home/linuxbrew/.linuxbrew/bin:$PATH"
+brew install $PACKAGE_NAME
+cp -Lr /home/linuxbrew/.linuxbrew/include/* ~/project/workspace/$artifact_name/include
+cp -Lr /home/linuxbrew/.linuxbrew/lib/*.so.* ~/project/workspace/$artifact_name/lib
```
Comment (varsill) [inline src/scripts/precompile_for_linux.sh:10]: I assume that there is no need to set the `rpath` in case of linux?

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:96f2fd20244cbffa294fa56d9c50dccb95eda730

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:1bd658cb2d654f7fd09add28c9afa3578b369695

File: src/scripts/precompile_for_linux.sh (hunk 2 above)
Comment (Noarkhh) [reply src/scripts/precompile_for_linux.sh]: yes

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:7315159f6714929143049b96a8acb602068da12a

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:b4aaaa1093c69b2dfc5ab958ac4f3c542a77b43d

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:2722881d216ecbb4e2e884a5f4e8056cb4e51e0c

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:57237a08c90fa5cf079f41fb5ff7c7ab1ed7c6b1

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:40b1cfd7fd143a88d056f5776a5080c73fc4459b

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:bb6e09188b2baef26dda6527b2a5e87a81b7f292

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:5db9d1338748dad57d9bbd342e980a6f2c76f712

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:1b74046e2173a377eb57edb267abc664d3bf9091

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:f059a80129a175c0b4e2d520bb686ef202d8a65b

File: src/jobs/precompile_for_macos_m1.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ parameters:
   expected-version:
     description: Version of the package
```
Comment (varsill) [inline src/jobs/precompile_for_macos_m1.yml]: I would add some more comprehensive description what does it mean that expected-version is `not a tag`.

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:f196d8f4809429df291e0a1634e4b9b870626d8e

**Review by varsill — APPROVED** [review]
🥇

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:a2d08be0febfe3e2acc93eacfec81091c23e6cfb

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:5cc867d568244c00e4faa3fe4cc5ab47d19da1cb

**Comment by bblaszkow06** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=dev:5cc867d568244c00e4faa3fe4cc5ab47d19da1cb

**Comment by bblaszkow06** [comment]
Your orb has been published to the CircleCI Orb Registry.
You can view your published orb on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/membraneframework/elixir?version=1.4.0

**Pull request merged**